src/Symfony/Component/EventDispatcher/DependencyInjection/RegisterListenersPass.php

```diff
@@ -108,7 +102,17 @@ public function process(ContainerBuilder $container)
                 throw new InvalidArgumentException(sprintf('Service "%s" must implement interface "%s".', $id, $interface));
             }
 
-            $definition->addMethodCall('addSubscriberService', array($id, $class));
+            foreach ($class::getSubscribedEvents() as $eventName => $params) {
+                if (is_string($params)) {
+                    $definition->addMethodCall('addListener', array($eventName, new ClosureProxyArgument(new Reference($id), $params)));
+                } elseif (is_string($params[0])) {
+                    $definition->addMethodCall('addListener', array($eventName, new ClosureProxyArgument(new Reference($id), $params[0]), isset($params[1]) ? $params[1] : 0));
+                } else {
+                    foreach ($params as $listener) {
+                        $definition->addMethodCall('addListener', array($eventName, new ClosureProxyArgument(new Reference($id), $listener[0]), isset($listener[1]) ? $listener[1] : 0));
+                    }
```

Comment:
is there a way to ensure that this logic stays in sync with the logic of the EventDispatcher to handle the return value of ``getSubscribedEvents`` ?